anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,407 @@
+        self.location = ensure_str(pkg.get("location", ""))
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "arch": self.arch,
+                "sourcepkg": self.source,
+                "size": self.size,
+                "metadata": self.metadata,
+                "location": self.location,
+            }
+        )
+        return result
+
+
+class BinaryHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, ensure_str(pkg.get("type", "binary").lower()))
+        self.license = ensure_str(pkg.get("license", ""))
+        self.files = pkg.get("files", [])
+        self.metadata = json.dumps(pkg.get("metadata", {}))
+        self.location = ensure_str(pkg.get("location", ""))
+
+    def validate_files(self):
+        for inp in [self.files]:
+            if type(inp) is not list:
+                raise HintsError(
+                    "bad hints record ({}), files, if specified must be list types".format(
+                        self.name
+                    )
+                )
+
+    def normalize(self):
+        self.validate_files()
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "files": self.files,
+                "metadata": self.metadata,
+                "location": self.location,
+            }
+        )
+        return result
+
+
+class DebianHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "dpkg")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+
+    def normalize(self):
+        if not self.source:
+            self.source = "%s-%s" % (self.name, self.version)
+        self.release = "N/A"
+
+    def to_dict(self) -> dict:
+        result = super().to_dict()
+        result.update(
+            {
+                "license": self.license,
+                "arch": self.arch,
+                "release": self.release,
+                "sourcepkg": self.source,
+                "size": self.size,
+            }
+        )
+        return result
+
+
+class AlpineHint(Hint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "APKG")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+        self.files = pkg.get("files", [])
+
+    def normalize(self):
+        if not self.release:
+            try:
+                (v, r) = self.version.split("-", 2)
```

Comment:
there is no need for the parentheses here
```suggestion
                v, r = self.version.split("-", 2)
```

Reply:
you could assign `self.release` and `self.version` right away, instead of using `v` and `r` and then assigning them later

Reply:
fixed